ompi/mca/hook/comm_method/hook_comm_method_fns.c

```diff
@@ -0,0 +1,788 @@
+/*
+ * Copyright (c) 2016-2018 IBM Corporation. All rights reserved.
+ * $COPYRIGHT$
+ *
+ * Additional copyrights may follow
+ *
+ * $HEADER$
+ */
+
+#include "ompi_config.h"
+
+#include "hook_comm_method.h"
+
+#ifdef HAVE_DLFCN_H
+#include <dlfcn.h>
+#endif
+
+#include "ompi/communicator/communicator.h"
+#include "ompi/mca/pml/pml.h"
+#include "opal/mca/btl/btl.h"
+#include "ompi/mca/pml/base/base.h"
+#include "ompi/mca/bml/base/base.h"
+#include "ompi/mca/mtl/base/base.h"
+
+// return the pml's module:component:name function pointer in fp
+static char*
+lookup_pml_name(void)
+{
+    return (char*) mca_pml_base_selected_component.pmlm_version.mca_component_name;
+}
+// return the mtl's module:component:name function pointer in fp
+static char*
+lookup_mtl_name(void)
+{
+    if (!ompi_mtl_base_selected_component) { return NULL; }
+    return (char*) ompi_mtl_base_selected_component->mtl_version.mca_component_name;
+}
+// Find the send btl's module:component:name for the incoming comm,rank
+static char*
+lookup_btl_name_for_send(ompi_communicator_t* comm, int rank) {
+    ompi_proc_t *dst_proc = ompi_comm_peer_lookup (comm, rank);
```

Comment:
Using this function forces the allocation of the underlying proc structure, but not it's initialization (so the endpoint will be NULL). I suggest using `ompi_group_peer_lookup_existing(comm->c_remote_group, rank)` instead, which will prevent the allocation of the proc if the proc is still the sentinel.